### PR TITLE
[error] "Better error message when vector used as if's condition variable (#1804)"

### PR DESCRIPTION
**Pull request by archibate**
Reverts taichi-dev/taichi#1808

Related PR = #1804

OFT fault for test failure: 2ffe504

### CODE REVIEW

**Review by Eydcao — APPROVED** [review]
Hi Yubin, the test results are good as what you desired in the previous fix. I did not notice there is any ci error as well. But I am not familiar with the workflow part so have to leave it for you guys to check.

So from what I can see, this is LGTM!

:>

File: docs/arithmetics.rst [hunk 1]
```diff
@@ -26,7 +26,7 @@ Arithmetic operators
     print(2 % 3)   # 2
     print(-2 % 3)  # 1
 
-  For C-style mod, please use ``ti.raw_mod``:
+  For C style mod, please use ``ti.raw_mod``:
```
Comment (yuanming-hu) [inline docs/arithmetics.rst:29]:
```suggestion
  For C-style mod, please use ``ti.raw_mod``:
```

File: python/taichi/lang/impl.py [hunk 2]
```diff
@@ -72,6 +72,17 @@ def begin_frontend_struct_for(group, loop_range):
     taichi_lang_core.begin_frontend_struct_for(group, loop_range.ptr)
 
 
+def begin_frontend_if(cond):
+    if is_taichi_class(cond):
+        raise ValueError(
+            'The truth value of vectors / matrices is ambiguous.\n'
```
Comment (yuanming-hu) [inline python/taichi/lang/impl.py:78]:
```suggestion
            'The truth value of vectors/matrices is ambiguous.\n'
```

File: python/taichi/lang/impl.py [hunk 3]
```diff
@@ -72,6 +72,17 @@ def begin_frontend_struct_for(group, loop_range):
...
+            'Consider using `any` or `all` when comparing vectors:\n'
```
Comment (yuanming-hu) [inline python/taichi/lang/impl.py:79]:
```suggestion
            'Consider using `any` or `all` when comparing vectors/matrices:\n'
```

File: docs/arithmetics.rst (hunk 1 above)
Comment (archibate) [reply docs/arithmetics.rst:29]: FYI L21: `The ``%`` operator in Taichi follows the Python style instead of C style`, should we change that too?

Comment (yuanming-hu) [reply docs/arithmetics.rst:29]: That one is fine.

**Pull request merged**